python/ray/serve/tests/test_http_state.py

```diff
@@ -1010,6 +1013,60 @@ def check_is_ready_for_shutdown():
     wait_for_condition(check_is_ready_for_shutdown)
 
 
+@patch("ray.serve._private.http_state.PROXY_READY_CHECK_TIMEOUT_S", 0.9)
+@pytest.mark.parametrize("number_of_worker_nodes", [1])
+def test_proxy_starting_timeout_longer_than_env(all_nodes, number_of_worker_nodes):
+    """Test calling update method on HTTPProxyState when the proxy state is STARTING and
+    when the ready call takes longer than PROXY_READY_CHECK_TIMEOUT_S.
+
+    The proxy state started with STARTING. After update is called, ready calls takes
+    some time to finish. The state will eventually change to HEALTHY after few more
+    tries.
+    """
+
+    @ray.remote(num_cpus=0)
+    class NewMockHTTPProxyActor:
```

Comment:
we should be able to unit test this behavior w/o a dependency on Ray and needing to start a ray instance)

could you try adding an interface for the proxy that abstracts details of ray.remote calls out? Examples to look at are in `test_replica_scheduler` and `test_deployment_state`. both essentially have actor fakes that are implemented for the tests